internal/pkg/workspace/workspace.go

```diff
@@ -109,79 +113,120 @@ func (ws *Workspace) Summary() (*archer.WorkspaceSummary, error) {
 		if err != nil {
 			return nil, err
 		}
-		wsSummary := archer.WorkspaceSummary{}
+		wsSummary := Summary{}
 		return &wsSummary, yaml.Unmarshal(value, &wsSummary)
 	}
 	return nil, &ErrNoProjectAssociated{}
 }
 
-func (ws *Workspace) writeSummary(projectName string) error {
-	summaryPath, err := ws.summaryPath()
+// AppNames returns the application names in the workspace.
+func (ws *Workspace) AppNames() ([]string, error) {
+	projectPath, err := ws.projectDirPath()
 	if err != nil {
-		return err
+		return nil, err
 	}
+	files, err := ws.fsUtils.ReadDir(projectPath)
+	if err != nil {
+		return nil, fmt.Errorf("read directory %s: %w", projectPath, err)
+	}
+	var names []string
+	for _, f := range files {
+		if !f.IsDir() {
+			continue
+		}
+		names = append(names, filepath.Base(f.Name()))
+	}
+	return names, nil
+}
 
-	workspaceSummary := archer.WorkspaceSummary{
-		ProjectName: projectName,
+// Read returns the contents of the file under the project directory joined by path elements.
+func (ws *Workspace) Read(elem ...string) ([]byte, error) {
+	projectPath, err := ws.projectDirPath()
+	if err != nil {
+		return nil, err
 	}
+	pathElems := append([]string{projectPath}, elem...)
+	return ws.fsUtils.ReadFile(filepath.Join(pathElems...))
+}
 
-	serializedWorkspaceSummary, err := yaml.Marshal(workspaceSummary)
+// Write writes the data to a file under the project directory joined by path elements.
+// If successful returns the full path of the file, otherwise returns an empty string and the error.
+func (ws *Workspace) Write(data []byte, elem ...string) (string, error) {
+	projectPath, err := ws.projectDirPath()
+	if err != nil {
+		return "", err
+	}
+	pathElems := append([]string{projectPath}, elem...)
+	filename := filepath.Join(pathElems...)
 
+	if err := ws.fsUtils.MkdirAll(filepath.Dir(filename), 0755 /* -rwxr-xr-x */); err != nil {
+		return "", fmt.Errorf("failed to create directories for file %s: %w", filename, err)
+	}
+	if err := ws.fsUtils.WriteFile(filename, data, 0644 /* -rw-r--r-- */); err != nil {
+		return "", fmt.Errorf("failed to write manifest file: %w", err)
+	}
+	return filename, nil
+}
+
+// DeleteApp removes the application directory from the project directory.
+func (ws *Workspace) DeleteApp(name string) error {
```

Comment:
I don't know if I still love this behavior. Maybe this isn't the PR to talk about it - but I'm a little scared of deleting someone's files. Even if we generated them. I think deleting the `.ecs-workspace` makes sense, since we'd not expect folks to mess with that. 

Anyway, just leaving this as a note to see what you think.

Reply:
yeah, I agree that we shouldn't ideally remove the local files. This way if users want to recreate the exact same pipeline or app then can just run `app deploy` instead of `app init` again.

My PR is already humongous :P we can change the behavior later.